core/src/test/java/dk/alexandra/fresco/framework/network/TestKryoNetNetwork.java

```diff
@@ -1,226 +1,328 @@
 package dk.alexandra.fresco.framework.network;
 
-import dk.alexandra.fresco.framework.MPCException;
 import dk.alexandra.fresco.framework.Party;
 import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
 import dk.alexandra.fresco.framework.configuration.NetworkConfigurationImpl;
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.After;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 public class TestKryoNetNetwork {
 
+  private static List<Exception> exceptions = new ArrayList<>();
+
+  @Before
+  public void reset() {
+    exceptions.clear();
+  }
+
+  /**
+   * Checks for exceptions and casts an Assert.fail if any was found
+   */
+  @After
```

Comment:
Taken from:
http://junit.org/junit4/javadoc/4.12/index.html?org/junit/After.html
"If you allocate external resources in a Before method you need to release them after the test runs. "

... not as a part of the test.
You should not fail in the after, it is intended for releasing external resources.